### PR TITLE
Hide demo card when showing demo from frontpage

### PR DESCRIPTION
**Pull request by balloob**
## Proposed change

When embedding the demo on the frontpage of HA, we add `?frontpage` to the query param. We will now hide the demo card when that is passed in.

Need to test this in CI demo build because I was unable to get `demo/script/develop_demo` working locally. It keeps hanging on requesting index.html (running from dev container)

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

## Summary by CodeRabbit

- **New Features**
  - Added a conditional "Welcome" section in the Home Assistant demo, which displays based on URL query parameters or specific utility checks.

### CODE REVIEW

**Comment by bramkragten** [comment]
I guess we now remove the query string because we navigate to a different url `https://demo.home-assistant.io/#/lovelace/home` also, should the string be kept when navigation to a different panel and back?

**Comment by bramkragten** [comment]
Maybe store it in `sessionStorage` before we do any navigation?

**Comment by balloob** [comment]
hmm is replacing the hash also replacing the query string?

**Comment by bramkragten** [comment]
> hmm is replacing the hash also replacing the query string?

yes

File: demo/src/configs/sections/lovelace.ts [hunk 1]
```diff
@@ -10,11 +10,12 @@ export const demoLovelaceSections: DemoConfig["lovelace"] = () => ({
...
-          title: "Welcome 👋",
-          cards: [{ type: "custom:ha-demo-card" }],
-        },
-        {
+          // @ts-expect-error
           cards: [
+            document.location.search !== "?frontpage" && {
+              title: "Welcome 👋",
+              cards: [{ type: "custom:ha-demo-card" }],
+            },
```
Comment (piitaya) [inline demo/src/configs/sections/lovelace.ts:18]:
```suggestion
            ...document.location.search !== "?frontpage" ? [{
              title: "Welcome 👋",
              cards: [{ type: "custom:ha-demo-card" }],
            }] : [],
```
So we don't need `ts-ignore` and filter Boolean.

File: demo/src/configs/sections/lovelace.ts [hunk 2]
```diff
@@ -65,7 +71,7 @@ export const demoLovelaceSections: DemoConfig["lovelace"] = () => ({
               entity: "media_player.living_room_nest_mini",
               name: "Nest Mini",
             },
-          ],
+          ].filter(Boolean),
```
Comment (piitaya) [inline demo/src/configs/sections/lovelace.ts:74]:
```suggestion
          ],
```

**Review by frenck — APPROVED** [review]

File: demo/src/entrypoint.ts [hunk 3]
```diff
@@ -1,4 +1,5 @@
 import "../../src/resources/safari-14-attachshadow-patch";
+import "./util/is_frontpage";
```
Comment (steverep) [inline demo/src/entrypoint.ts:2]: AI is correct here.  This import is doing nothing since the module has no side effects.

Comment (balloob) [reply demo/src/entrypoint.ts:2]: It is working as expected though. The query string is stripped off by Lovelace and by evaluating early, we capture it.

Comment (steverep) [reply demo/src/entrypoint.ts:2]: Oh right, forgot about that.  Sorry noticed while fixing conflict for #21247.

**Pull request merged**